Rebus.RabbitMq/Config/RabbitMqQueueOptionsBuilder.cs

```diff
@@ -26,14 +27,30 @@ public RabbitMqQueueOptionsBuilder SetExclusive(bool exclusive)
         }
 
         /// <summary>
-        /// Set auto delete
+        /// Set auto delete, when last consumer disconnects
         /// </summary>
         public RabbitMqQueueOptionsBuilder SetAutoDelete(bool autoDelete)
```

Comment:
Super! 👍 One thing though: I think the purpose of the TTL would be more clear, if it was added to the `.SetAutoDelete(...)` method signature.

It could look like this:
`public RabbitMqQueueOptionsBuilder SetAutoDelete(bool autoDelete, long ttlInMs = 0)`

What do you think?

Reply:
I think personally think that's more than reasonable. I only kept them apart because I didnt know if there were any major fundamental differences between the two things. RabbitMQ documentation does state the two options can be used in conjunction with one another, I just assumed if you were setting a ttl, the queue deletion was kind of implied.

I'll get the pull request updated with the suggestions there. Thanks!

Reply:
One question I had about this, I'm depending on Arguments collection not being overwritten with `SetArguments` later. Should set arguments be something more of update/insert instead of just re-assigning entire arguments dictionary?

Reply:
> Should set arguments be something more of update/insert instead of just re-assigning entire arguments dictionary?

Yes, I think that would cause the least surprises.

Reply:
> are you suggesting combining `setAutoDelete` and `setTtl` correct

Yes 😄

Reply:
I went ahead and submitted the changes for the signature combination. I did not include the changes for the setArguments as I think simply setting/overriding settings in arguments would work, but potentially leave you no way to null out values, remove values all together....

I considered something like the following, but wanted your take on it before submitting that yet. I can create another pull request.
``` cs
public RabbitMqQueueOptionsBuilder SetArguments(Dictionary<string, object> arguments)
        {
            if(arguments != null)
            {
               arguments.ForEach(kvp => {
                  Arguments[kvp.Key] = kvp.Value;
               });
            } else {
               Arguments.Clear();
            }
            return this;
        }
```

or maybe a a parameter that explicitly clears out the arguments dictionary prior to setting the values.